src/Sylius/Component/Core/OrderProcessing/OrderPaymentProcessor.php

```diff
@@ -62,7 +62,7 @@ public function process(BaseOrderInterface $order)
             return;
         }
 
-        $newPayment = $order->getLastNewPayment();
+        $newPayment = $order->getLastCartPayment();
```

Comment:
But is ``OrderPaymentProcessor`` called after placing order? For now, it would be called on each checkout step (which is the same as we had it before) and it's also called on ``fail`` and ``cancel`` transition on order payment.